zipline/gens/tradesimulation.py

```diff
@@ -185,6 +189,9 @@ def handle_benchmark(date):
                 elif action == DAY_START:
                     once_a_day(dt)
                 elif action == DAY_END:
+                    if algo.data_frequency == 'minute':
```

Comment:
the `data_frequency` isn't changing for the lifetime of this simulation, so we shouldn't pay the cost of checking this (even though it's daily and not minutely, so the perf impact isn't huge).  also, it's a bit misleading to future readers of this code.

one way to get around this is something like this, above the main loop:
```
if algo.data_frequency == "minute":
    def execute_cancelation_policy():
        algo.blotter.execute_cancel_policy(DAY_END)
else:
    def execute_cancelation_policy():
        pass
```

and then call `execute_cancelation_policy()` inside the `DAY_END` handler.